apps/web-giddh/src/app/purchase/purchase-order/purchase-order.component.ts

```diff
@@ -95,10 +96,28 @@ export class PurchaseOrderComponent implements OnInit, OnDestroy {
     public showSelectAllItemCheckbox: boolean = false;
     /* Send email request params object */
     public sendEmailRequest: any = {};
-
-    constructor(private modalService: BsModalService, private generalService: GeneralService, private breakPointObservar: BreakpointObserver, public purchaseOrderService: PurchaseOrderService, private store: Store<AppState>, private toaster: ToasterService, public route: ActivatedRoute) {
+    /* Observable for filters applied */
+    public purchaseOrderListFilters$: Observable<any>;
+    /* This will hold if we need to overwriter filters */
+    public useStoreFilters: boolean = false;
+    /* This will hold current page url */
+    public pageUrl: string = "pages/purchase-management/purchase-orders";
+
+    constructor(private modalService: BsModalService, private generalService: GeneralService, private breakPointObservar: BreakpointObserver, public purchaseOrderService: PurchaseOrderService, private store: Store<AppState>, private toaster: ToasterService, public route: ActivatedRoute, private router: Router, public purchaseOrderActions: PurchaseOrderActions) {
         this.activeCompanyUniqueName$ = this.store.pipe(select(state => state.session.companyUniqueName), (takeUntil(this.destroyed$)));
-        this.universalDate$ = this.store.select(state => state.session.applicationDate).pipe(takeUntil(this.destroyed$));
+        this.universalDate$ = this.store.pipe(select(state => state.session.applicationDate), takeUntil(this.destroyed$));
+        this.purchaseOrderListFilters$ = this.store.pipe(select(state => state.purchaseOrder.listFilters), (takeUntil(this.destroyed$)));
+
+        this.router.events.pipe(takeUntil(this.destroyed$)).subscribe(event => {
```

Comment:
Please move these new subscriptions to ngOnInit method to avoid component instantiation